core/ts/MaciState.ts

```diff
@@ -549,47 +560,42 @@ class Poll {
             this.copyStateFromMaci()
         }
 
-        const deactivatedKeyHash = hash3([...deactivatedPublicKey.asArray(), salt]);
-        const deactivatedKeyIndex = this.deactivatedKeyEvents.findIndex(d => d.keyHash === deactivatedKeyHash);
+        const deactivatedKeyHash: BigInt = hash3([...deactivatedPublicKey.asArray(), salt]);
+        const deactivatedKeyIndex = this.deactivatedKeyEvents.findIndex(d => d.keyHash.toString() == deactivatedKeyHash.toString());
 
         if (deactivatedKeyIndex === -1) {
-            console.log("Key index is -1");
+            console.log("Deactivated key is missing from the deactivated key events collection from contract");
             return {};
         }
 
         const deactivatedKeyEvent = this.deactivatedKeyEvents[deactivatedKeyIndex];
 
-        const stateIndexInt = parseInt(stateIndex.toString());
-        const computedStateIndex = stateIndexInt > 0 && stateIndexInt <= this.numSignUps ? stateIndexInt - 1 : -1;
-        const currentStateLeaves = [this.stateLeaves[computedStateIndex]];
-
         const z = BigInt(42);
 
         const [c1r, c2r] = elGamalRerandomize(
-            newPublicKey.rawPubKey,
+            this.coordinatorKeypair.pubKey.rawPubKey,
```

Comment:
Should be `this.coordinatorKeypair.pubKey` (?)

Could you please check that you are passing the correct value?

Reply:
this.coordinatorKeypair.pubKey is a class PubKey from DomainObejcts.
elGamalRerandomize expects type PubKey frm Crypto, which is just a BigInt[] and which coresponds to the this.coordinatorKeypair.pubKey.rawPubKey

Reply:
> this.coordinatorKeypair.pubKey is a class PubKey from DomainObejcts. elGamalRerandomize expects type PubKey frm Crypto, which is just a BigInt[] and which coresponds to the this.coordinatorKeypair.pubKey.rawPubKey

Thanks for the explanation! :+1: